pandas-stubs/core/groupby/groupby.pyi

```diff
@@ -176,7 +176,10 @@ class GroupBy(BaseGroupBy[NDFrameT]):
     @overload
     def all(self: GroupBy[DataFrame], skipna: bool = ...) -> DataFrame: ...
     @final
-    def count(self) -> NDFrameT: ...
+    @overload
+    def count(self: GroupBy[Series]) -> Series[int]: ...
+    @overload
+    def count(self: GroupBy[DataFrame]) -> DataFrame: ...
```

Comment:
I'd prefer removing `def count()` from here, and having declarations in `DataFrameGroupBy` and `SeriesGroupBy` in `core/groupby/generic.pyi`

Reply:
> I'm a new contributor to this repo and eager to learn more about these choices. Is there a specific reason for moving the declarations to `core/groupby/generic.pyi`?
> 
> In the actual pandas repo, the `count()` method is implemented in [`core/groupby/groupby.py`](https://github.com/pandas-dev/pandas/blob/v2.2.2/pandas/core/groupby/groupby.py), not [`core/groupby/generic.py`](https://github.com/pandas-dev/pandas/blob/v2.2.2/pandas/core/groupby/generic.py).
> 
> Thanks in advance for taking the time to explain.

It probably is due to that we had issues in the past with doing overloads where `self` was declared as `Series` vs. `DataFrame`, so putting the stubs in the lower classes was a way to get around that problem.

The type checkers have improved, so it's possible your approach could be used going forward.

Having said that, I see that for `any()` and `all()` in the file you changed, that was the pattern used, so maybe your change was fine the way it was!

As an example, there are a number of methods shared between `Series` and `DataFrame` that are implemented in `NDFrame`, but instead of putting the stub in `NDFrame`, we put them in `Series` and `DataFrame` to clarify that the results are different in the two subclasses.